src/Controller/Action/ExportSelectedToCsvAction.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\WishlistPlugin\Controller\Action;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Sylius\Bundle\OrderBundle\Factory\AddToCartCommandFactoryInterface;
+use Sylius\Component\Core\Factory\CartItemFactoryInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Order\Context\CartContextInterface;
+use Sylius\Component\Order\Modifier\OrderItemQuantityModifierInterface;
+use Sylius\WishlistPlugin\Command\Wishlist\ExportWishlistToCsv;
+use Sylius\WishlistPlugin\Command\Wishlist\WishlistItem;
+use Sylius\WishlistPlugin\Entity\WishlistInterface;
+use Sylius\WishlistPlugin\Repository\WishlistRepositoryInterface;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Messenger\Stamp\HandledStamp;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+final readonly class ExportSelectedToCsvAction
+{
+    public function __construct(
+        private WishlistRepositoryInterface $wishlistRepository,
+        private CartContextInterface $cartContext,
+        private AddToCartCommandFactoryInterface $addToCartCommandFactory,
+        private CartItemFactoryInterface $cartItemFactory,
+        private OrderItemQuantityModifierInterface $orderItemQuantityModifier,
+        private MessageBusInterface $messageBus,
+        private UrlGeneratorInterface $urlGenerator,
+        private TranslatorInterface $translator,
+    ) {
+    }
+
+    public function __invoke(string $wishlistId, Request $request, Session $session): BinaryFileResponse|RedirectResponse
+    {
+        /** @var ?WishlistInterface $wishlist */
+        $wishlist = $this->wishlistRepository->find((int) $wishlistId);
+        if (null === $wishlist) {
+            $session->getFlashBag()->add('error', $this->translator->trans('sylius_wishlist_plugin.ui.wishlist_not_exists'));
+
+            return new RedirectResponse($this->urlGenerator->generate('sylius_wishlist_plugin_shop_locale_wishlist_list_wishlists'));
+        }
+
+        $selectionJson = (string) $request->request->get('selection', '[]');
+        $indices = $this->decodeSelection($selectionJson);
+        $wishlistName = trim((string) $request->request->get('wishlistName', 'wishlist'));
```

Comment:
Is the wishlist name not included in the wishlist itself? Why is the value taken from the request?